### PR TITLE
[rescheduling] Add sanity check

### PR DESCRIPTION
**Pull request by BenjaminLudwigSAP**

### CODE REVIEW

File: octavia_f5/api/drivers/f5_driver/tasks/reschedule_tasks.py [hunk 1]
```diff
@@ -121,3 +123,19 @@ def revert(self, result, load_balancer: models.LoadBalancer, candidate: str, rem
...
         self._loadbalancer_repo.update(db_apis.get_session(), load_balancer.id, server_group_id=removal_host)
+
+
+class SanityCheck(RescheduleTasks):
+    def execute(self, loadbalancer_id: str, candidate: str):
+        """Check the rescheduling parameters for sanity."""
+
+        # Check that target host exists.
+        # Else the LB would get deleted, but not recreated anywhere.
+        if not self._amphora_repo.get_devices_for_host(candidate):
```
Comment (notandy) [inline octavia_f5/api/drivers/f5_driver/tasks/reschedule_tasks.py]: get_devices_for_host takes two parameters, not one: session and host.
I'd recommend you to add easy unit tests to your PRs.

**Review by notandy — CHANGES_REQUESTED** [review]

**Review by notandy — CHANGES_REQUESTED** [review]
Could you split of the refactoring in a seperate PR, thanks.

**Comment by BenjaminLudwigSAP** [comment]
Yeah, I was already contemplating doing that.

**Comment by BenjaminLudwigSAP** [comment]
Replacing this PR with #216, which will be implemented via separate PRs.

**Pull request closed without merging**